engine/ddf/src/ddfc.py

```diff
@@ -307,9 +330,25 @@ def to_cxx_descriptor(context, pp_cpp, pp_h, message_type, namespace_lst):
         if '"' in default:
             pp_cpp.p('char DM_ALIGNED(4) %s_%s_%s_DEFAULT_VALUE[] = %s;', namespace, message_type.name, f.name, default)
 
+    oneof_scope = None
     lst = []
     for f in message_type.field:
-        tpl = (f.name, f.number, f.type, f.label)
+        one_of_index = 0
+
+        if f.HasField("oneof_index"):
+            oneof_decl = message_type.oneof_decl[f.oneof_index]
+
+            # indices start at zero, but we need to distinguish between which fields belong
+            # to a oneof block somehow
+            one_of_index = f.oneof_index + 1

```

Comment:
Isn't that the "m_OneOfSet" member?
Since we're doing struct initialixation on line 377, sholdn't it be its own variable?
E.g. number, type and label also are bitfields and are set explicitly.